client/finality-grandpa/src/environment.rs

```diff
@@ -1178,6 +1182,19 @@ pub(crate) fn finalize_block<BE, Block, Client>(
 			},
 		};
 
+		// Q: We `finalized()` this at L37, so can I be sure
+		// that it's fine to unwrap here?
+		if let Some(justification) = justification.clone() {
```

Comment:
The code above needs to be changed, otherwise we only get the encoded blob of the justification here.